source/core/Ky.ts

```diff
@@ -217,11 +217,17 @@ export class Ky {
 				throw error;
 			}
 
-			const retryAfter = error.response.headers.get('Retry-After');
+			const retryAfter = error.response.headers.get('Retry-After')
+				?? error.response.headers.get('RateLimit-Reset')
+				?? error.response.headers.get('X-RateLimit-Reset') // GitHub
+				?? error.response.headers.get('X-Rate-Limit-Reset'); // Twitter
 			if (retryAfter && this._options.retry.afterStatusCodes.includes(error.response.status)) {
 				let after = Number(retryAfter) * 1000;
 				if (Number.isNaN(after)) {
 					after = Date.parse(retryAfter) - Date.now();
+				} else if (after >= Date.parse('2024-01-01')) {
```

Comment:
Why is this hardcoded to "2024-01-01"?

Reply:
It is stupid, yes.

See the comment on the next line (the linter would not let me put it above).

Some `RateLimit-Reset` implementations in the wild use delta seconds, while others use timestamps. Both are integers, with no easy way to disambiguate them other than, essentially, how large the number is. Generally speaking, we would expect delta seconds to be a small number (because it's referring to seconds _from now_) and a timestamp to be a large number (because it's referring to seconds _from epoch_). Ky wants to normalize these values to delta seconds. So we need to detect timestamps and convert them.

The most obvious approach might be to check if `after` is close to `Date.now()`, but that is susceptible to clock skew and incorrectly set clocks. Such bugs also have a history of leading to security vulnerabilities. So, we use a constant threshold instead.

How large should `after` be for us to consider it a timestamp? That's... very arbitrary. @fregante suggested we use `1700000000` in this comment: https://github.com/sindresorhus/ky/pull/618#discussion_r1713150598

`1700000000` milliseconds is a little less than 20 days, which should be fine in the vast majority of cases. But if someone wants to set a 6 month retry delay and hope the computer stays up for that long... who am I to stop them? I figured that `Date.parse('2024-01-01')`, being the beginning of the year in which this feature was implemented, is hopefully more readable, allows for extremely long retry delays, and makes it obvious that the value is arbitrary and not based on some standard (I don't think there _is_ a standard for this).

I'm very open to improvements here, suggestions welcome! Would be nice to put it in a `const` with a self-documenting name, but I couldn't think of a good name.

Reply:
> `1700000000` milliseconds is a little less than 20 days

`Retry-After` uses seconds, not milliseconds. Since this value is meant to differentiate that header from an epoch-based value, this is fine. No server will ask you to retry [in 53 years](https://www.wolframalpha.com/input?i=1700000000+seconds) (unless you're somehow querying [Deep Thought](https://en.wikipedia.org/wiki/List_of_The_Hitchhiker%27s_Guide_to_the_Galaxy_characters#Deep_Thought))


https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

Reply:
`Retry-After` and `RateLimit-Reset` both use seconds ([except for Mulesoft](https://docs.mulesoft.com/gateway/latest/policies-included-rate-limiting-sla), apparently 😡).

However, we [convert them](https://github.com/sindresorhus/ky/blob/a7204c4672d10a144b12ef169c8d38e17db81931/source/core/Ky.ts#L225) both to milliseconds. Indeed, `1700000000` seconds is plenty high enough. I could've just added a few more zeros for milliseconds. I just thought `Date.parse('2024-01-01')` (which is `1704067200` seconds) had a certain logic to it. Perfectly happy to change it, though.